RecoLocalCalo/HcalRecAlgos/src/HcalQIECodersGPU.cc

```diff
@@ -0,0 +1,69 @@
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalQIECodersGPU.h"
+
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+HcalQIECodersGPU::HcalQIECodersGPU(HcalQIEData const& qiedata)
+    : totalChannels_{qiedata.getAllContainers()[0].second.size() + qiedata.getAllContainers()[1].second.size()},
+      offsets_(totalChannels_ * numValuesPerChannel),
+      slopes_(totalChannels_ * numValuesPerChannel) {
+  auto const containers = qiedata.getAllContainers();
+
+  // fill in hb
+  auto const& barrelValues = containers[0].second;
+  for (uint64_t i = 0; i < barrelValues.size(); ++i) {
+    for (uint32_t k = 0; k < 4; k++)
```

Comment:
This "4" appears quite often. Since you defined `numValuesPerChannel` (with a `static constexpr`), perhaps in RecoLocalCalo/HcalRecAlgos/interface/HcalQIECodersGPU.h you could have more conveniently defined
```C++
  static constexpr uint32_t numSomething = 4;
  static constexpr uint32_t numValuesPerChannel = numSomething*numSomething;
```
and then replace "4" with `numSomething` where needed